src/internal/FontWeightEditor/FontWeightEditor.tsx

```diff
@@ -0,0 +1,170 @@
+    { value: '200' },
+    { value: '300' },
+    { value: '400' },
+    { value: '500' },
+    { value: '600' },
+    { value: '700' },
+    { value: '800' },
+    { value: '900' },
+    { value: 'normal' },
+    { value: 'bold' },
+    { value: 'bolder' },
+    { value: 'lighter' },
+    { value: 'inherit' },
+    { value: 'initial' },
+    { value: 'unset' },
+];
+
+function FontWeightEditor({ value, onChange }: TProps) {
+    const [localValue, setLocalValue] = React.useState<TFontWeightValue | null>({ value: `${value}` as TFontWeight });
+
+    const [errorMessage, setErrorMessage] = React.useState<string>('');
+    const classes = useStyles();
+
+    const handleValidation = (value: string): string => {
+        const valueIsNumber = +value;
+        if (valueIsNumber) {
+            const acceptableNumbers = valueIsNumber > 0 && valueIsNumber <= 1000;
+            return !acceptableNumbers ? 'Please enter numbers in this range [1,1000]' : '';
+        } else {
+            const acceptableValue = fontWeightValues.find((item) => item.value === value);
+            return !acceptableValue ? 'Please enter valid font weight value' : '';
+        }
+    };
+
+    const handleChange = useCallback(
+        (event, newValue) => {
+            setErrorMessage('');
+            if (typeof newValue === 'string') {
+                const validationError = handleValidation(newValue);
+                if (validationError) {
+                    return setErrorMessage(validationError);
+                }
+
+                setValue({
+                    value: newValue,
+                });
+            } else if (newValue?.inputValue) {
+                const validationError = handleValidation(newValue.inputValue);
+                if (validationError) {
+                    return setErrorMessage(validationError);
+                }
+
+                // Create a new value from the user input
+                setValue({
+                    value: newValue.inputValue,
+                });
+            } else {
+                if (!newValue?.value) {
+                    return;
+                }
+                const validationError = handleValidation(newValue.value);
+                if (validationError) {
+                    return setErrorMessage(validationError);
+                }
+                setValue(newValue);
+            }
+        },
+        [onChange]
+    );
+
+    const setValue = useCallback(
+        (fontWeight) => {
+            clearTimeout(timer);
+            timer = null;
+
+            timer = setTimeout(() => {
+                timer = null;
+                onChange(fontWeight.value);
+            }, 50);
+
+            setLocalValue(fontWeight);
+        },
+        [onChange]
+    );
+
+    useEffect(() => {
+        // Update localValue if value is changed from outside(ex. reset)
+        if (timer === null && value !== localValue.value) {
+            setLocalValue({ value: `${value}` as TFontWeight });
```

Comment:
this `as` will not be needed anymore
```suggestion
            setLocalValue({ value });
```